webtasks/cron_backend_mongodb.js

```diff
@@ -319,6 +330,71 @@ router.put('/:container/:name',
             .then(res.json.bind(res), next);
 });
 
+router.put('/:container/:name/state',
+    ensure('headers', ['host']),
+    ensure('params', ['container', 'name']),
+    function (req, res, next) {
+        var validStates = ['active', 'inactive'];
+        var data = req.webtaskContext.data;
+        var jobs = req.mongo.collection(data.JOB_COLLECTION);
+        var cluster_host = data.CLUSTER_HOST;
+        var state = req.body.state || req.query.state;
+        
+        if (validStates.indexOf(state) < 0) {
+            return next(Boom.badRequest('A job\'s state can only be set to one '
+                + 'of `' + validStates.join('`, `') + '`.'));
+        }
+        
+        var query = {
+            cluster_url: cluster_host,
+            container: req.params.container,
+            name: req.params.name,
+        };
+
+        var projection = {
+            results: { $slice: 1 },
+        };
+    
+        Bluebird.promisify(jobs.findOne, jobs)(query, projection)
+            .catch(function (err) {
+                throw Boom.wrap(err, 503, 'Error querying database.');
+            })
+            .then(function (job) {
+                if (!job) {
+                    throw Boom.notFound('No such job `'
+                        + cluster_host + '/api/cron/'
+                        + req.params.container + '/'
+                        + req.params.name + '`.');
+                }
+    
+                return job;
+            })
+            .then(function (job) {
+                if (validStates.indexOf(job.state) < 0) {
+                    return next(Boom.preconditionFailed('The job is in an invalid state.'));
+                }
+                
+                // Make sure that no intervening action changed the job's state.
+                query.state = job.state;
```

Comment:
Presumably the "expired" state is set by CRON when it comes around to running the job and finds out the token is no longer valid?

If so, wouldn't the system be eventually consistent even if we allow that race condition to run its course? Do we care about stronger assurances?